<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FrmFTP</class>
 <widget class="QMainWindow" name="FrmFTP">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>657</width>
    <height>489</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>657</width>
    <height>489</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>657</width>
    <height>489</height>
   </size>
  </property>
  <property name="palette">
   <palette>
    <active>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>20</red>
        <green>22</green>
        <blue>24</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>88</red>
        <green>88</green>
        <blue>88</blue>
       </color>
      </brush>
     </colorrole>
    </active>
    <inactive>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>20</red>
        <green>22</green>
        <blue>24</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>88</red>
        <green>88</green>
        <blue>88</blue>
       </color>
      </brush>
     </colorrole>
    </inactive>
    <disabled>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>88</red>
        <green>88</green>
        <blue>88</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>88</red>
        <green>88</green>
        <blue>88</blue>
       </color>
      </brush>
     </colorrole>
    </disabled>
   </palette>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowIcon">
   <iconset theme="QIcon::ThemeIcon::FolderDragAccept"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>636</width>
      <height>76</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: 1px solid #224075;
</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Shadow::Raised</enum>
    </property>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>18</y>
       <width>101</width>
       <height>18</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none</string>
     </property>
     <property name="text">
      <string>ram offset:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtStartAddress">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>40</y>
       <width>80</width>
       <height>26</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Consolas</family>
      </font>
     </property>
     <property name="text">
      <string>0000</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="opt1Ram">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>10</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none</string>
     </property>
     <property name="text">
      <string>To RAM</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="opt1File">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>10</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none</string>
     </property>
     <property name="text">
      <string>To FILE</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="opt1FileOverwrite">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>10</y>
       <width>121</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none</string>
     </property>
     <property name="text">
      <string>File Overwrite</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="cmdCloseFTP">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>450</y>
      <width>86</width>
      <height>26</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>117</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Light">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>215</red>
          <green>235</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Midlight">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>236</red>
          <green>56</green>
          <blue>56</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Highlight">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>0</green>
          <blue>4</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Accent">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>0</green>
          <blue>4</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>117</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Light">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>215</red>
          <green>235</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Midlight">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>236</red>
          <green>56</green>
          <blue>56</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Highlight">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>0</green>
          <blue>4</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Accent">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>0</green>
          <blue>4</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>117</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Light">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>215</red>
          <green>235</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Midlight">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>236</red>
          <green>56</green>
          <blue>56</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Accent">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>0</green>
          <blue>4</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="text">
     <string>[ close ]</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="chkClearLogAuto">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>450</y>
      <width>171</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear on new send.</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdClearLog">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>450</y>
      <width>101</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>clear log</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>21</x>
      <y>2</y>
      <width>105</width>
      <height>18</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="text">
     <string> Target Location:</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>94</y>
      <width>636</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: 1px solid #224075;
</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Shadow::Raised</enum>
    </property>
    <widget class="QRadioButton" name="opt2Chunk1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>100</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none</string>
     </property>
     <property name="text">
      <string>256 bytes</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="opt2Chunk2">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>10</y>
       <width>100</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none</string>
     </property>
     <property name="text">
      <string>512 bytes</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="opt2Chunk3">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>10</y>
       <width>100</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none</string>
     </property>
     <property name="text">
      <string>1k bytes</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="opt2Chunk4">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>10</y>
       <width>100</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none</string>
     </property>
     <property name="text">
      <string>2k bytes</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="opt2Chunk5">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>10</y>
       <width>100</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none</string>
     </property>
     <property name="text">
      <string>4k bytes</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>86</y>
      <width>74</width>
      <height>18</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="text">
     <string> Chunk size </string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdClearRam">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>50</y>
      <width>80</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear ram &gt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdSendFile">
    <property name="geometry">
     <rect>
      <x>224</x>
      <y>50</y>
      <width>71</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Send &gt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdRunInRam">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>50</y>
      <width>50</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Run !</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdViewRam">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>50</y>
      <width>60</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>view</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="chkAutoRun">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>52</y>
      <width>84</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>auto run</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdBrowseFile">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>150</y>
      <width>86</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Browse...</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdHIDController">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>450</y>
      <width>111</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>HID Ctrl...</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>-30</x>
      <y>140</y>
      <width>750</width>
      <height>3</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLineEdit" name="txtFileselected">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>150</y>
      <width>540</width>
      <height>26</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Consolas</family>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">border: 1px solid #224075;
</string>
    </property>
   </widget>
   <widget class="QProgressBar" name="pbFileupload">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>181</y>
      <width>480</width>
      <height>24</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QProgressBar {
    border: 1px solid #3c6dc3;    /* blue outer border */
    border-radius: 4px;
    text-align: center;
    background-color: #2b2b2b;    /* dark empty bar */
    color: #ffffff;                /* text color */
}

QProgressBar::chunk {
    background-color: #3c6dc3;    /* blue fill */
    border: 1px solid #2b2b2b;    /* match empty bar for nice border effect */
    border-radius: 3px;            /* slightly smaller than bar radius */
    margin: 0px;
}
</string>
    </property>
    <property name="value">
     <number>25</number>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Horizontal</enum>
    </property>
    <property name="invertedAppearance">
     <bool>false</bool>
    </property>
    <property name="format">
     <string>%p%</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblProgress">
    <property name="geometry">
     <rect>
      <x>502</x>
      <y>183</y>
      <width>140</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>00:00 (0.0 KB/s)</string>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>-30</x>
      <y>210</y>
      <width>750</width>
      <height>3</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Horizontal</enum>
    </property>
   </widget>
   <widget class="QTextEdit" name="txtFTPLog">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>637</width>
      <height>221</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="maximumSize">
     <size>
      <width>16777215</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>26</red>
          <green>31</green>
          <blue>63</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>26</red>
          <green>31</green>
          <blue>63</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled/>
     </palette>
    </property>
    <property name="font">
     <font>
      <family>Consolas</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::WinPanel</enum>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Consolas'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Liberation Mono';&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cmdAbort">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>50</y>
      <width>71</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>abort!</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>54</y>
      <width>58</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>0x</string>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
